src/core/shape_inference/include/reduce_shape_inference.hpp

```diff
@@ -1,16 +1,42 @@
 // Copyright (C) 2018-2023 Intel Corporation
 // SPDX-License-Identifier: Apache-2.0
 //
+
 #pragma once
-#include <openvino/core/validation_util.hpp>
-#include <openvino/op/util/arithmetic_reductions_keep_dims.hpp>
-#include <openvino/op/util/logical_reduction_keep_dims.hpp>
-#include <openvino/opsets/opset1.hpp>
 
+#include "openvino/core/validation_util.hpp"
+#include "openvino/op/util/arithmetic_reductions_keep_dims.hpp"
+#include "openvino/op/util/logical_reduction_keep_dims.hpp"
 #include "utils.hpp"
 
 namespace ov {
 namespace op {
+namespace util {
+
+template <class T>
+result_shape_t<T> reduce_shape(const T& input_shape, std::vector<int64_t>& axes, const bool keep_dims) {
```

Comment:
Is it possible to reuse common reduce helper here, or to use shape_infer in the evaluate?

Reply:
It give not good result in term of binary size. Current solution is sufficient in evaluate.